packages/vite/src/node/ssr/ssrTransform.ts

```diff
@@ -31,11 +31,21 @@ export async function ssrTransform(
 ): Promise<TransformResult | null> {
   const s = new MagicString(code)
 
-  const ast = parser.parse(code, {
-    sourceType: 'module',
-    ecmaVersion: 'latest',
-    locations: true
-  }) as any
+  let ast
+  try {
+    ast = parser.parse(code, {
+      sourceType: 'module',
+      ecmaVersion: 'latest',
+      locations: true
+    }) as any
+  } catch (err) {
+    const line = err.loc.line
+    throw new Error(
+      `Parse failure: ${err.message}\nContents of line ${line}: ${
+        code.split('\n')[line - 1]
```

Comment:
If there happens any kind of other error, an error would be thrown in the catch.
So we should check if error really contains loc and loc.line, also to make it typescript safe.

Reply:
Ok, I added a check for `err.loc` and `err.loc.line`